src/main/java/org/elasticsearch/common/lucene/all/AllTermQuery.java

```diff
@@ -51,7 +51,7 @@ public AllTermQuery(Term term) {
     }
 
     @Override
-    public Weight createWeight(IndexSearcher searcher, boolean needsScores) throws IOException {
+    public SpanWeight createWeight(IndexSearcher searcher, boolean needsScores) throws IOException {
```

Comment:
Maybe you are a bit behind. I added tests for this position check last night and fixed bugs in the default impl.